azurerm/resource_arm_streamanalytics_job.go

```diff
@@ -835,104 +834,104 @@ func parseArmStreamAnalyticsOutputDatasource(data map[string]interface{}) (strea
 		return &result, nil
 
 	case string(streamanalytics.TypeMicrosoftServiceBusEventHub):
-		if data["service_bus_namespace"] == nil {
-			return nil, fmt.Errorf("Event Hub output %s missing required field service_bus_namespace", name)
+		namespace, err := extractAndValidateRequiredProperty(data, "service_bus_namespace")
+		if err != nil {
+			return nil, err
 		}
-		namespace := data["service_bus_namespace"].(string)
-
-		if data["eventhub_name"] == nil {
-			return nil, fmt.Errorf("Event Hub output %s missing required field eventhub_name", name)
+		eventHub, err := extractAndValidateRequiredProperty(data, "eventhub_name")
+		if err != nil {
+			return nil, err
 		}
-		eventhub := data["eventhub_name"].(string)
-
-		if data["shared_access_policy_name"] == nil {
-			return nil, fmt.Errorf("Event Hub output %s missing required field shared_access_policy_name", name)
+		policyName, err := extractAndValidateRequiredProperty(data, "shared_access_policy_name")
+		if err != nil {
+			return nil, err
 		}
-		policyName := data["shared_access_policy_name"].(string)
-
-		if data["shared_access_policy_key"] == nil {
-			return nil, fmt.Errorf("Event Hub output %s missing required field shared_access_policy_key", name)
+		policyKey, err := extractAndValidateRequiredProperty(data, "shared_access_policy_key")
+		if err != nil {
+			return nil, err
 		}
-		policyKey := data["shared_access_policy_key"].(string)
+		partitionKey := extractOptionalProperty(data, "eventhub_partition_key")
 
 		result := streamanalytics.EventHubOutputDataSource{
 			Type: streamanalytics.TypeMicrosoftServiceBusEventHub,
 			EventHubOutputDataSourceProperties: &streamanalytics.EventHubOutputDataSourceProperties{
-				ServiceBusNamespace:    &namespace,
-				SharedAccessPolicyName: &policyName,
-				SharedAccessPolicyKey:  &policyKey,
-				EventHubName:           &eventhub,
+				ServiceBusNamespace:    namespace,
+				SharedAccessPolicyName: policyName,
+				SharedAccessPolicyKey:  policyKey,
+				EventHubName:           eventHub,
+				PartitionKey:		partitionKey,
 			},
 		}
 
-		if v := data["eventhub_partition_key"]; v != nil {
-			value := v.(string)
-			result.EventHubOutputDataSourceProperties.PartitionKey = &value
-		}
-
 		return &result, nil
 
 	case string(streamanalytics.TypeMicrosoftSQLServerDatabase):
-		// TODO Validate fields
-		databaseServer := data["database_server"].(string)
-		databaseName := data["database_name"].(string)
-		databaseTable := data["database_table"].(string)
-		databaseUser := data["database_user"].(string)
-		databasePassword := data["database_password"].(string)
+		databaseServer, err := extractAndValidateRequiredProperty(data, "database_server")
```

Comment:
If go supports currying/partial application, this is a good opportunity to use it. The data parameter could be partially applied. Not sure if go can though.

Reply:
@rickerbh - have made updates.